File: src/SolarTally.Domain/ValueObjects/TimeInterval.cs
using System;
using System.Collections.Generic;
using Ardalis.GuardClauses;
using SolarTally.Domain.Common;

namespace SolarTally.Domain.ValueObjects
{
    public class TimeInterval : ValueObject
{
    public TimeSpan Start { get; private set; }

    public TimeSpan End { get; private set; }

    public TimeSpan Difference { get; private set; }

    private TimeInterval() { }

    public TimeInterval(int startHr, int startMin, int endHr, int endMin)
    {
        Guard.Against.InvalidTimeInterval(startHr, startMin, endHr, endMin);
        var start = new TimeSpan(startHr, startMin, 0);
        var end = new TimeSpan(endHr, endMin, 0);
        Start = start;
            End = end;
            Difference = End - Start;
        }

        protected override IEnumerable<object> GetAtomicValues()
        {
            // Using a yield return statement to return each element one at a time
            yield return Start;
            yield return End;
            yield return Difference;
        }
    }

    public class TimeIntervalAbrv
    {
        public TimeSpan Start { get; set; }
        public TimeSpan End { get; set; }
    }
}

File: src/SolarTally.WebUI_Ng/ClientApp/src/app/shared/dtos/time-interval.ts
import { TimeSpan, TimeSpanAbrv } from './time-span';

export class TimeInterval
{
    start: TimeSpan;
    end: TimeSpan;
    difference: TimeSpan;
}

export class TimeIntervalAbrv
{
    start: TimeSpanAbrv;
    end: TimeSpanAbrv;
    difference: TimeSpanAbrv;
}

export class TimeIntervalStr
{
    start: string;
    end: string;
    difference: string;
}